Sources/ErgastAPI/Endpoint.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  Endpoint.swift
+//  
+//
+//  Created by Giovanni Noa on 4/28/20.
+//
+
+import Foundation
+
+/// Object to use for generating a URL with a given Path enum case..
+struct Endpoint {
+    private var path = "/api/f1"
+    
+    
+    /// Initializer for an Endpoint object.
+    /// - Parameters:
+    ///   - path: Specify a path, mapping to a specific endpoint of the Ergast REST API.
+    ///   - season: Specify either all historical circuits, or circuits for a specific year.
+    init(with path: Path, for season: Season?) {
```

Comment:
If this can be initialized with a `nil` season then consider adding a default value of nil:
```swift
init(with path: Path, for season: Season?=nil)
```
That way later you can call it like:
```swift
Endpoint(with: .circuits)
```
instead of
```swift
Endpoint(with: .circuits, for: nil)
```
which is a little nicer.